src/parser/csql_grammar.y

```diff
@@ -5185,45 +5192,125 @@ only_all_class_spec
 		DBG_PRINT}}
 	;
 
-class_name
-	: identifier DOT identifier
+user_specified_name_without_dot
+	: identifier_without_dot DOT identifier_without_dot
 		{{
 
-			PT_NODE *user_node = $1;
-			PT_NODE *name_node = $3;
+			PT_NODE *user = $1;
+			PT_NODE *name = $3;
 
-			if (name_node != NULL && user_node != NULL)
+			if (user
+			    && name
+			    && user->node_type == PT_NAME
+			    && name->node_type == PT_NAME)
 			  {
-			    name_node->info.name.resolved = pt_append_string (this_parser, NULL,
-			                                                      user_node->info.name.original);
+			    name->info.name.resolved = pt_append_string (this_parser, NULL, user->info.name.original);
+
+			    parser_free_tree (this_parser, user);
+
+			    // PT_NAME_INFO_SET_FLAG (name, PT_NAME_INFO_USER_SPECIFIED);
+			  }
+
+			$$ = name;
+			PARSER_SAVE_ERR_CONTEXT ($$, @$.buffer_pos)
+
+		DBG_PRINT}}
+	| identifier_without_dot
+		{{
+
+			PT_NODE *name = $1;
+
+			if (name
+			    && name->node_type == PT_NAME)
+			  {
+			    // PT_NAME_INFO_SET_FLAG (name, PT_NAME_INFO_USER_SPECIFIED);
 			  }
-			if (user_node != NULL)
+
+			$$ = name;
+			PARSER_SAVE_ERR_CONTEXT ($$, @$.buffer_pos)
+
+		DBG_PRINT}}
+	;
+
+user_specified_name
+	: identifier DOT identifier
+		{{
+
+			PT_NODE *user = $1;
+			PT_NODE *name = $3;
+
+			if (user
+			    && name
+			    && user->node_type == PT_NAME
+			    && name->node_type == PT_NAME)
 			  {
-			    parser_free_tree (this_parser, user_node);
+			    name->info.name.resolved = pt_append_string (this_parser, NULL, user->info.name.original);
+
+			    parser_free_tree (this_parser, user);
+			    
+			    // PT_NAME_INFO_SET_FLAG (name, PT_NAME_INFO_USER_SPECIFIED);
 			  }
 
-			$$ = name_node;
+			$$ = name;
 			PARSER_SAVE_ERR_CONTEXT ($$, @$.buffer_pos)
 
 		DBG_PRINT}}
 	| identifier
 		{{
 
-			$$ = $1;
+			PT_NODE *name = $1;
+
+			if (name
+			    && name->node_type == PT_NAME)
+			  {
+			    // PT_NAME_INFO_SET_FLAG (name, PT_NAME_INFO_USER_SPECIFIED);
+			  }
+
+			$$ = name;
 			PARSER_SAVE_ERR_CONTEXT ($$, @$.buffer_pos)
 
 		DBG_PRINT}}
 	;
 
-class_name_list
-	: class_name_list ',' class_name
+user_specified_name_list
+	: user_specified_name_list ',' user_specified_name
 		{{
 
 			$$ = parser_make_link($1, $3);
 			PARSER_SAVE_ERR_CONTEXT ($$, @$.buffer_pos)
 
 		DBG_PRINT}}
-	| class_name
+	| user_specified_name
+		{{
+
+			$$ = $1;
+			PARSER_SAVE_ERR_CONTEXT ($$, @$.buffer_pos)
+
+		DBG_PRINT}}
+	;
+
+object_name
```

Comment:
What is the difference between user_specified_name and object_name?
It looks identical.
If there is no reason to divide the replacement for the existing class_name into two, it is better to unify it into one object_name.
Or wouldn't it be better to keep class_name as it is?

Reply:
user_specified_name sets the PT_NAME_INFO_USER_SPECIFIED flag to change the original name to user_name.object_name before semantic check.
```
"PT_NAME_INFO_SET_FLAG(name, PT_NAME_INFO_USER_SPECIFIED);"
```
```
class_name → user_specified_name_without_dot (PT_NAME_INFO_USER_SPECIFIED)
class_name → user_specified_name (PT_NAME_INFO_USER_SPECIFIED)
class_name_list → user_specified_name_list (PT_NAME_INFO_USER_SPECIFIED)
class_name → object_name
```

Reply:
OK